python/ml4ir/base/model/cyclic_learning_rate.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Cyclical Learning Rate Schedule policies for TensorFlow."""
+
+import tensorflow as tf
```

Comment:
@mohazahran I also want us to add a FIXME / TODO comment to remove this class when we move to tf2.3 and add tf_addons as a dependency.

Reply:
With that said, I'm quite happy that this trick works out of the box.

Reply:
neat!